lib/common/src/DevMode.php

```diff
@@ -62,12 +71,23 @@ public static function hasExemptionForNode(DOMNode $node)
      */
     public static function isExemptFromValidation(DOMNode $node)
     {
-        $document = $node instanceof Document ? $node : $node->ownerDocument;
+        $document = self::getDocument($node);
+        return self::isActiveForDocument($document) && self::hasExemptionForNode($node);
+    }
 
-        if (! $document) {
-            return false;
+    /**
+     * Get the document from the specified node.
+     *
+     * @param DOMNode $node
+     * @return Document|DOMDocument|null
```

Comment:
Won't this only ever return a `Document` instance? The `Document::fromNode()` method only returns a `Document`.

```suggestion
     * @return Document
```